packages/gui/src/components/addressbook/ContactEdit.tsx

```diff
@@ -0,0 +1,342 @@
+// import { useGetNFTInfoQuery } from '@chia-network/api-react';
+import { fromBech32m } from '@chia-network/api';
+import { AddressBookContext, Flex, Form, TextField } from '@chia-network/core';
+import { Trans } from '@lingui/macro';
+import { Add, Remove } from '@mui/icons-material';
+import { Box, Button, IconButton, Typography } from '@mui/material';
+import React, { useContext } from 'react';
+import { useForm, useFormContext, useFieldArray } from 'react-hook-form';
+import { useParams, useNavigate } from 'react-router-dom';
+
+// import { launcherIdFromNFTId } from '../../util/nfts';
+// import NFTPreview from '../nfts/NFTPreview';
+
+function AddressFields() {
+  const { control } = useFormContext();
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'addresses',
+  });
+
+  const handleAppend = (value) => {
+    append(value);
+  };
+
+  const handleRemove = (index) => {
+    remove(index);
+  };
+
+  return (
+    <>
+      <Box display="flex" alignItems="center" justifyContent="space-between">
+        <Typography variant="h6">
+          <Trans>Addresses</Trans>
+        </Typography>
+        <IconButton onClick={() => handleAppend({ name: '', address: '' })}>
+          <Add />
+        </IconButton>
+      </Box>
+      {fields.map((item, index) => (
+        <Box key={item.id} display="flex" alignItems="center" justifyContent="space-between" gap={2}>
+          <TextField
+            render={({ field }) => <input {...field} />}
+            defaultValue={item.name}
+            name={`addresses[${index}].name`}
+            control={control}
+            variant="filled"
+            color="secondary"
+            fullWidth
+            disabled={false}
+            label={<Trans>Name</Trans>}
+          />
+          <TextField
+            render={({ field }) => <input {...field} />}
+            defaultValue={item.address}
+            name={`addresses[${index}].address`}
+            control={control}
+            variant="filled"
+            color="secondary"
+            fullWidth
+            disabled={false}
+            label={<Trans>Address</Trans>}
+          />
+          <IconButton onClick={() => handleRemove(index)}>
+            <Remove />
+          </IconButton>
+        </Box>
+      ))}
+    </>
+  );
+}
+
+function ProfileFields() {
+  const { control } = useFormContext();
+  const { fields, append, remove } = useFieldArray({
+    control,
```

Comment:
you can skip control, it will be used automatically

Reply:
Tracked here: https://github.com/Chia-Network/issue-tracker/issues/536